### PR TITLE
should fix issue where 'read on closed body' error was leaking down

### PR DESCRIPTION
**Pull request by whyrusleeping**
This should address the issues seen here: https://travis-ci.org/ipfs/go-ipfs/jobs/73239014#L3661

I had special cased the error in one spot earlier where this error is always hit, but in cases where we pass the reader back directly, its a bit of a race on whether the final read happens before or after the body is closed. This embeds the error check into the reader wrapping the body so we always get that error checked properly and converted into an io.EOF

License: MIT
Signed-off-by: Jeromy <jeromyj@gmail.com>

### CODE REVIEW

File: commands/http/client.go [hunk 1]
```diff
@@ -305,6 +306,11 @@ type httpResponseReader struct {
 
 func (r *httpResponseReader) Read(b []byte) (int, error) {
 	n, err := r.resp.Body.Read(b)
+
+	// reading on a closed response body is as good as an io.EOF here
+	if err != nil && strings.Contains(err.Error(), "read on closed response body") {
+		err = io.EOF
+	}
```
Comment (jbenet) [inline commands/http/client.go]: @zignig was this what you were seeing? `"read on closed response body"`

**Comment by jbenet** [comment]
@whyrusleeping this LGTM :+1: any known way to repro the error? (for a test case)

**Comment by jbenet** [comment]
LGTM but not to travis. looks like a lot of bad luck. rerunning

**Comment by whyrusleeping** [comment]
@jbenet yeah, i couldnt figure out how to get it to repro on my computer sadly.

**Pull request merged**